### ISSUE TITLE
Exclude some upstream servers from cache_optimistic

### ISSUE THREAD

**Issue by YBS-PC** [AdguardTeam/AdGuardHome#7022]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to request a feature or enhancement and not ask a question

### The problem

I have to use smartdns to geo-unblock some domains, and for this smartdns issues a limited set of ip addresses as responses to DNS requests. For example, when requesting gemini google, I can get a response on 230.200.100.001 and gemini google will think that I am in the USA, but when the ttl of the installed smartdns ends, this ip will be used for another site.

### Proposed solution

Use the field to exclude some upstream DNS servers from the optimistic cache.

### Alternatives considered and additional information

config file

**Comment by ainar-g** [comment]
There is no separate optimistic cache, only the optimistic mode for the cache.  As such, this sounds like a duplicate of #4197.
